### ISSUE TITLE
Has the `setNeutral` method been removed?

### ISSUE THREAD

**Issue by gr2m** [actions/toolkit#146]
I don't find it in the code. And there is a test that claims to test it but actually does not:

https://github.com/actions/toolkit/blob/6fcaac504651656d8ae9f23ff94b386925ed9281/packages/core/__tests__/lib.test.ts#L128-L131

Has the neutral exit code been removed from actios altogether?

**Comment by peter-evans** [comment]
> Has the neutral exit code been removed from actios altogether?

My guess is that Actions doesn't handle other exit codes yet and so this functionality is not yet available. The test seems to have been added like that. This is when it was first commit.
https://github.com/actions/toolkit/commit/a526749ce96e66c4e8c32ddc43f099c6830d0345#diff-ce52518b05efe344d702cb7e7d01da6aR38-R41

**Comment by ann-kilzer** [comment]
I would really love a neutral exit code.

**Comment by bryanmacfarlane** [comment]
👋 the service has a GA item to respect neutral code.  The toolkit doesn't implement.  That test was a copy past that just tests failure.

A bit on history - actions v2 has a status of skipped.  It was asserted that's what we needed.  However, skipped is more about conditions on steps and jobs where it never executes (your action code never enters).  Neutral is more about you got into the action code and the code has determined that work is not relevant.  

There's another effort around smarter triggers / conditions etc. to ensure compute is not allocated just to short circuit for common cases (branches, paths etc.).

I'm going to mark this external (waiting on service changes) and enhancement (since there is not real toolkit path) and then follow up here.

Hope that makes sense.

**Comment by gr2m** [comment]
Is the exit code 78 still supported or should we refrain from using it at this point?

**Comment by bryanmacfarlane** [comment]
Neutral (78) is not respected right now but there's plans to add before GA.  There's discussions around whether it's 78.  This toolkit was waiting on that before adding the method.

@ethomson

**Comment by ethomson** [comment]
> Is the exit code 78 still supported or should we refrain from using it at this point?

I'd refrain from using it.  I'd like to switch to some unambiguous out-of-band mechanism for signalling neutral continuation.  (eg, `::set-status neutral`)

**Comment by JaKXz** [comment]
This action uses `exit 78` in its [entrypoint](https://github.com/veggiemonk/skip-commit/issues/2#issuecomment-550460284), but, how would one accomplish something like `::set-status neutral` to stop the following actions? Is there a `::set-status pass` or skip or something?

**Comment by ericsciple** [comment]
@JaKXz it looks like the readme was updated to include the example to use `job.<job_id>.if` condition to skip a job

note, steps also allow `if` conditions.

**Comment by ericsciple** [comment]
@bryanmacfarlane is https://github.community/t5/GitHub-Actions/bd-p/actions the correct place for this request?

**Comment by ericsciple** [comment]
looks like here is the github community issue https://github.community/t5/GitHub-Actions/GitHub-Actions-quot-neutral-quot-exit-code-is-incorrectly/td-p/29051

**Comment by miigotu** [comment]
I have a situation where I need to mark a specific job as "not required for overall pass of the run"

I have a matrix of different operating systems and different python versions for my build. In this build, I also have an experimental marker, so I have for example:

python3.10, ubuntu-latest, experimental=False
python3.11-dev, ubuntu-latest, experimental=True

I want to require all runs to succeed for pull requests, and I also want to be able to show "build: passing" on my readme and the ❌ or ✅ next to the commits to be green.

I set `fail_fast: matrix.experimental` so my runs happily will fail on experimental job without interrupting the rest of my build, and I expect that specific set of steps to fail. This currently sets my whole run as I unsuccessful for purpose of the commit build statuses and or bud statuses. 

How can I make something this work, where I can have my cake and eat it too? I don't feel like writing tests and actions for software that is still into development so that my application is ready the second it is released is a bad thing, and that NOT doing that is less secure and short sighted. I was hoping to be able to mark my job as neutral when it is an experimental build failing (expected failure like in unit test), and success when the build magically starts working due to things like lxml starting to build wheels for python 3.11 so that I could send a sweet message of success (telegram, discord, email, whatever) that my application now works on python 3.11.

It would be amazing if we could have a flag adjacent to fail_fast, such as `expected_result: [zero, nonzero] # default zero` and possibly `failure_status: [success, fail, neutral] # default fail`

That solves the whole problem for me. I'd be willing to help if its something the maintainers thought useful.
